lib/chat_app_web/live/single_room_live.ex

```diff
@@ -66,6 +85,24 @@ defmodule ChatAppWeb.SingleRoomLive do
     end))}
   end
 
+  def handle_info({:users_typing, username, status}, socket) do
+    bool_as_int = fn x -> if x, do: 1, else: 0 end
+    int_as_bool = fn x -> if x == 0, do: false, else: true end
```

Comment:
No need for these 2 functions.
To check the current status of users typing, all you need is to check if `username` is a key in the `users_typing` map (or an entry in the list, if you go that way)
And the animation condition can be written like this:
```elixir
animation_condition = !!Bitwise.xor(current_status, sent_status)
```
to "cast" it to a boolean.

Side note: whenever you have some function that goes like this:
```elixir
def is_positive(x) do
    if x > 0, do: true, else: false
```
you can rewrite it like this:
```elixir
def is_positive(x), do: x > 0
```
in general, we use `if/else` if its branches would return non-boolean values, as in:
```elixir
if x > 0, do: :positive, else: :negative
```

Reply:
There is no xor function in the Bitwise module, only bxor which works on integers. I've tried your suggestion but it doesn't work, unfortunately.